ipam/azure_test.go

```diff
@@ -0,0 +1,192 @@
+	"		</IPSubnet>" +
+	"	</Interface>" +
+	"</Interfaces>"
+
+// Handles queries from IPAM source.
+func handleIpamQuery(w http.ResponseWriter, r *http.Request) {
+	w.Write([]byte(ipamQueryResponse))
+}
+
+func parseResult(stdinData []byte) (*cniTypesCurr.Result, error) {
+	result := &cniTypesCurr.Result{}
+	if err := json.Unmarshal(stdinData, result); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+func TestAzure(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Azure source Suite")
+}
+
+var (
+	_ = Describe("Test azure source", func() {
+
+		var (
+			testAgent *common.Listener
+			source *azureSource
+			err error
+		)
+
+		BeforeSuite(func() {
+			// Create a fake local agent to handle requests from IPAM plugin.
+			u, _ := url.Parse("tcp://" + ipamQueryUrl)
+			testAgent, err = common.NewListener(u)
+			Expect(err).NotTo(HaveOccurred())
+
+			testAgent.AddHandler("/", handleIpamQuery)
+
+			err = testAgent.Start(make(chan error, 1))
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		AfterSuite(func() {
+			// Cleanup.
+			testAgent.Stop()
+		})
+
+		Describe("Test create Azure source", func() {
+
+			Context("When create new azure source with empty options", func() {
+				It("Should return as default", func() {
+					options := make(map[string]interface{})
+					source, err = newAzureSource(options)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(source.name).Should(Equal("Azure"))
+					Expect(source.queryUrl).Should(Equal(azureQueryUrl))
+					Expect(source.queryInterval).Should(Equal(azureQueryInterval))
+				})
+			})
+
+			Context("When create new azure source with options", func() {
+				It("Should return with default queryInterval", func() {
+					options := make(map[string]interface{})
+					second := 7
+					queryInterval := time.Duration(second) * time.Second
+					queryUrl := "http://testqueryurl:12121/test"
+					options[common.OptIpamQueryInterval] = second
+					options[common.OptIpamQueryUrl] = queryUrl
+					source, err = newAzureSource(options)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(source.name).Should(Equal("Azure"))
+					Expect(source.queryUrl).Should(Equal(queryUrl))
+					Expect(source.queryInterval).Should(Equal(queryInterval))
+				})
+			})
+		})
+
+		Describe("Test Azure source refresh", func() {
+			Context("Create source for testing refresh", func() {
+				It("Should create successfully", func() {
+					options := make(map[string]interface{})
+					options[common.OptEnvironment] = common.OptEnvironmentAzure
+					options[common.OptAPIServerURL] = "null"
+					options[common.OptIpamQueryUrl] = "http://"+ipamQueryUrl
+					source, err = newAzureSource(options)
+					Expect(err).ShouldNot(HaveOccurred())
+					Expect(source.name).Should(Equal("Azure"))
+					Expect(source.queryUrl).Should(Equal("http://"+ipamQueryUrl))
+					Expect(source.queryInterval).Should(Equal(azureQueryInterval))
+				})
+			})
+
+			Context("When time interval is less", func() {
```

Comment:
couldnot understand what this test for..what we are testing here

Reply:
Neither can I. Let's remove this test unless you can make it more descriptive and please also remember test will also be maintained as part of our code.

Reply:
I will change the description as "refresh interval is too short". It may be more clearly now.